### PR TITLE
Mock data

### PR DESCRIPTION
**Pull request by charliearlie**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,49 +1,53 @@
...
+    "@types/react-dom": "16.8.4",
+    "axios": "^0.18.0",
+    "react": "^16.8.6",
+    "react-dom": "^16.8.6",
+    "react-scripts": "3.0.0",
+    "typescript": "3.4.5"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "start:mocks": "concurrently \"json-server --watch -p 7777 src/api/mocks/db.json\" \"REACT_APP_USE_MOCKS=true react-scripts start\"",
```
Comment (charliearlie) [inline package.json]: Only new thing here

File: src/components/Drawer/SideDrawer.tsx [hunk 2]
```diff
@@ -27,11 +29,24 @@ const styles = (theme: Theme) => ({
...
 
 function SideDrawer(props: Props) {
-  const { classes, items } = props;
+  const { classes } = props;
+  const [drawerItems, setDrawerItems] = useState<any>([]);
+
+  function fetchDrawerItems() {
+    axios(Endpoints.drawerItems).then((res) => {
+      if (res) {
+        setDrawerItems(res.data);
```
Comment (charliearlie) [inline src/components/Drawer/SideDrawer.tsx]: async await wasn't working for some odd reason?

File: src/components/Drawer/DrawerItem.tsx [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
+import { withStyles } from '@material-ui/core/styles';
+import { WithStyles } from '@material-ui/core';
+import { Theme } from '@material-ui/core/styles/createMuiTheme';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import Icon from '@material-ui/core/Icon';
 
-export type drawerItem = {
+export interface IDrawerItem {
```
Comment (LorenzoKoundouris) [inline src/components/Drawer/DrawerItem.tsx]: Ideally interfaces should be in their own file, but not terribly important. That way it be a default export and avoid `{ .. }` when imported.

File: src/components/Drawer/SideDrawer.tsx [hunk 4]
```diff
@@ -27,11 +29,24 @@ const styles = (theme: Theme) => ({
 });
 
 interface Props extends WithStyles<typeof styles> {
-  items: Array<drawerItem>;
+  items?: Array<IDrawerItem>;
```
Comment (LorenzoKoundouris) [inline src/components/Drawer/SideDrawer.tsx]: A convention I follow is `let arr: IDrawerItem[]` for single typed arrays and `let arr: Array<IDrawerItem | IDrawerItem2>`. Let me know what your thoughts are on this approach

File: src/components/Drawer/SideDrawer.tsx [hunk 5]
```diff
@@ -27,11 +29,24 @@ const styles = (theme: Theme) => ({
...
 }
 
 function SideDrawer(props: Props) {
-  const { classes, items } = props;
+  const { classes } = props;
+  const [drawerItems, setDrawerItems] = useState<any>([]);
+
+  function fetchDrawerItems() {
+    axios(Endpoints.drawerItems).then((res) => {
```
Comment (LorenzoKoundouris) [inline src/components/Drawer/SideDrawer.tsx]: Missing return type and argument type for `res`

File: src/components/Drawer/SideDrawer.tsx [hunk 6]
```diff
@@ -27,11 +29,24 @@ const styles = (theme: Theme) => ({
...
+      if (res) {
```
Comment (LorenzoKoundouris) [inline src/components/Drawer/SideDrawer.tsx]: Perhaps validate `res.data` here

File: src/components/Drawer/SideDrawer.tsx [hunk 7]
```diff
@@ -27,11 +29,24 @@ const styles = (theme: Theme) => ({
...
+        setDrawerItems(res.data);
+      }
+    });
+  }
+
+  useEffect(() => {
```
Comment (LorenzoKoundouris) [inline src/components/Drawer/SideDrawer.tsx]: Missing return type

File: src/components/Drawer/DrawerItem.tsx (hunk 3 above)
Comment (charliearlie) [reply src/components/Drawer/DrawerItem.tsx]: I just think as it's a drawer item it's quite relevant to be in there.

File: src/components/Drawer/SideDrawer.tsx (hunk 4 above)
Comment (charliearlie) [reply src/components/Drawer/SideDrawer.tsx]: Literally have no idea what you're on about hahaha

File: src/components/Drawer/SideDrawer.tsx (hunk 5 above)
Comment (charliearlie) [reply src/components/Drawer/SideDrawer.tsx]: Done

File: src/components/Drawer/SideDrawer.tsx (hunk 6 above)
Comment (charliearlie) [reply src/components/Drawer/SideDrawer.tsx]: But if res doesn't exist we'll get "cannot read data of undefined", no?

File: src/components/Drawer/SideDrawer.tsx (hunk 7 above)
Comment (charliearlie) [reply src/components/Drawer/SideDrawer.tsx]: Looked at best practices for this and none involve adding a return type, so not sure what to do with it

**Pull request merged**